packages/web-app-files/tests/unit/components/SideBar/FileInfo.spec.js

```diff
@@ -0,0 +1,184 @@
+import { createLocalVue, mount } from '@vue/test-utils'
+import FileInfo from '@files/src/components/SideBar/FileInfo.vue'
+
+import GetTextPlugin from 'vue-gettext'
+
+const localVue = createLocalVue()
+localVue.use(DesignSystem)
+localVue.use(Vuex)
+localVue.use(GetTextPlugin, {
+  translations: 'does-not-matter.json',
+  silent: true
+})
+
+const filesPersonalRoute = { name: 'files-personal' }
+const filesPublicRoute = { name: 'files-public-list' }
+
+function getWrapper(
+  route,
+  { filename = 'testfile', extension, type = 'file', publicLink = false }
+) {
+  const mountStubs = { ...stubs, 'oc-button': false }
+  const mountOptions = {
+    localVue,
+    store: createStore({}, filename, extension, type),
+    stubs: mountStubs,
+    mocks: {
+      $route: route,
+      publicPage: () => publicLink
+    },
+    propsData: {
+      isContentDisplayed: true
+    }
+  }
+  return mount(FileInfo, mountOptions)
+}
+
+const selectors = {
+  fileNameItem: '.file_info h2',
+  favoritesButton: '.file_info button.file_info__favorite',
+  fileDescription: '.file_info__body div'
+}
+
+describe('SideBar', () => {
+  describe('when user on personal route', () => {
+    it('should show file name', () => {
+      const wrapper = getWrapper(filesPersonalRoute, {
+        filename: 'testfile',
+        extension: 'jpg'
+      })
+
+      const fileName = wrapper.find(selectors.fileNameItem)
+      expect(fileName.isVisible()).toBeTruthy()
+      expect(fileName.text()).toBe('testfile.jpg')
+    })
+
+    it('should show favorites button in files personal page', () => {
+      const wrapper = getWrapper(filesPersonalRoute, {
+        filename: 'testfile',
+        extension: 'jpg'
+      })
+
+      const favoritesButton = wrapper.find(selectors.favoritesButton)
+      expect(favoritesButton.exists()).toBeTruthy()
+    })
+
+    it('should not show favorites button in public link page', () => {
+      const wrapper = getWrapper(filesPublicRoute, {
+        filename: 'testfile',
+        extension: 'jpg',
+        publicLink: true
+      })
+
+      const favoritesButton = wrapper.find(selectors.favoritesButton)
+      expect(favoritesButton.exists()).toBeFalsy()
+    })
+
+    it('favorites button should work', async () => {
+      const wrapper = getWrapper(filesPersonalRoute, {
+        filename: 'testfile',
+        extension: 'jpg'
+      })
+
+      const favoriteTrigger = jest.fn()
+      wrapper.vm.markFavorite = favoriteTrigger
+      const favoritesButton = wrapper.find(selectors.favoritesButton)
+      await favoritesButton.trigger('click.native.stop')
+      expect(favoriteTrigger).toHaveBeenCalledTimes(1)
+      const args = favoriteTrigger.mock.calls[0]
+      expect(args.length).toBe(1)
+      expect(args[0]).toMatchObject({
+        client: undefined,
+        file: {
+          id: '4',
+          fileId: '4',
+          icon: 'file',
+          name: 'testfile.jpg',
+          extension: 'jpg',
```

Comment:
do you have to provide the `extension` in the name? 🤔 
```suggestion
          name: 'testfile',
          extension: 'jpg',
```